neuralmonkey/nn/mlp.py

```diff
@@ -1,55 +1,37 @@
 # tests: lint, mypy
 
 import tensorflow as tf
-from neuralmonkey.nn.projection import linear, nonlinear
 
 
-def multilayer_perceptron(input_, layer_sizes, activation=tf.tanh,
-                          dropout_plc=None, scope="mlp"):
-    mlp_input = input_
-
-    with tf.variable_scope(scope):
-        for i, size in enumerate(layer_sizes):
-            mlp_input = nonlinear(mlp_input, size, activation=activation,
-                                  scope="mlp_layer_{}".format(i))
-            if dropout_plc:
-                mlp_input = tf.nn.dropout(mlp_input, dropout_plc)
-
-    return mlp_input
-
-
-# TODO is the class below equivalent/worse/better than the function above?
-# Should we deprecate this class?
 class MultilayerPerceptron(object):
     """ General implementation of the multilayer perceptron. """
 
     # pylint: disable=too-many-arguments
     def __init__(self, mlp_input, layer_configuration, dropout_plc,
-                 output_size, name='multilayer_perceptron'):
+                 output_size, activation_fn=tf.nn.relu,
+                 name='multilayer_perceptron'):
 
         with tf.variable_scope(name):
             last_layer = mlp_input
             last_layer_size = mlp_input.get_shape()[1].value
 
             self.n_params = 0
             for i, size in enumerate(layer_configuration):
-                last_layer = tf.tanh(
-                    linear(last_layer, size,
-                           scope="dense_layer_{}".format(i + 1)))
+                last_layer = tf.contrib.layers.fully_connected(
```

Comment:
me prijde ze kusy kodu, ktery kontrolujou mraky lidi v tensowflow bychom od nich meli sourcovat ... to je hlavni duvod proc jsem to nahradil ... v puvodni verzy byla napriklad zadratovana tanh, coz nemluve o tom, ze relu je lepsi default, nepovoluje menit si tu funkci (ano dalo by se to predelat v nasi verzi, ale pak pribudou dalsi veci, ktere treba tahle fully_connected ma a nemusime vynalezat kolo)